### ISSUE TITLE
Changes in group organisation structure

### ISSUE THREAD

**Issue by NotSpecial** [amiv-eth/amivapi#97]
In contrast to events, which have several usecases of events which have participants which are not AMIV members, I cannot think of any case in which some random person should enter a group/mailinglist on their own.

Current situation:

Groups have user and address members. If the field "public" is true, address members are allowed.
Registered users can sign up for any group.

While this structure was fitting for mailinglists, I think it needs to be changed for groups.

We need groups which allow no self enroll (e.g. Vorstand)
We dont need the possibility for random people to enter any group (I believe)
But we need the possibility to add additional email addresses to a group (for the purpose of forwarding mails to other groups or external systems, e.g. lan-party which needs to reach VIS as well)

I would propose the following changes:

the "public" field controls if AMIV Members can sign up themselves to the group, e.g. a "Helfer" group.
If it is not public, only the group owner or admins can add people to groups.
GroupAddressMembers could be renamed to "AdditionalAdress" or something, They can only be added by group owner/admin.

So basically: no random people for groups anymore, because not needed.

Any thoughts on this?

**Comment by Leonidaz0r** [comment]
All your proposed changes are implemented already, however there should be improvements to the code structure. I opened a separate issue #98
